src/stdatamodels/fits_support.py

```diff
@@ -775,7 +775,8 @@ def callback(node, json_id):
             data = hdulist[pair].data
             return data
         return node
-    af.tree = treeutil.walk_and_modify(af.tree, callback)
+    # don't assign to af.tree to avoid an extra validation
+    af._tree = treeutil.walk_and_modify(af.tree, callback)
```

Comment:
Changing this assignment has 2 benefits.
1) (required for this PR) the modified tree returned from `walk_and_modify` contains `FITS_rec` instances (this function, `_map_hdulist_to_arrays` is responsible for replacing the arrays referenced in the tree with the data from the hdus which might come back as `FITS_rec` instances). If `af.tree` was assigned, asdf would attempt to validate the assigned tree which would result in warnings (in asdf 3.0) if the tree contained `FITS_rec` instances
2) The assignment to `af.tree` results in validation of the tree. This is unnecessary as this function (`_map_hdulist_to_arrays`) is called right after a call to `asdf.open` (see [from_fits_asdf](https://github.com/spacetelescope/stdatamodels/blob/7c2cb0a55caee6b5282ef7a8c22e9d6de7d51a9f/src/stdatamodels/fits_support.py#L743-L751) in the same file) which by default will validate the tree on read. Assigning to `af._tree` avoids this duplicate validation.